src/main/java/org/jenkinsci/plugins/p4/scm/BranchesScmSource.java

```diff
@@ -97,10 +113,47 @@ public P4Browser getBrowser() {
 		return list;
 	}
 
+	@Override
+	public Workspace getWorkspace(List<P4Path> paths) {
+		P4Path branchPath = Iterables.getFirst(paths, null);
+		if(branchPath == null){
+			throw new IllegalArgumentException("missing branch path");
+		}
+
+		String client = getFormat();
+		String mappingFormat = String.format("%1s/%%1$s //%2$s/%%1$s", branchPath.getPath(), client);
+
+		StringBuffer workspaceView = new StringBuffer(1024);
+		workspaceView.append(String.format(mappingFormat, getScriptPathOrDefault("Jenkinsfile")));
+		for(String mapping : getViewMappings()){
+			workspaceView.append("\n").append(String.format(mappingFormat, mapping));
+		}
+		WorkspaceSpec spec = new WorkspaceSpec(workspaceView.toString(), null);
+		return new ManualWorkspaceImpl(getCharset(), false, client, spec);
+	}
+
+	private String getScriptPathOrDefault(String defaultScriptPath) {
```

Comment:
I moved getScriptPathOrDefault to the Abstract class AbstractP4ScmSource; to save duplication could you change it to protected in AbstractP4ScmSource and remove it from here.